engine/access/rest/request_decorator.go

```diff
@@ -0,0 +1,140 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"github.com/onflow/flow-go/engine/access/rest/middleware"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// a convenience wrapper around the http request to make it easy to read request query params
+type requestDecorator struct {
+	*http.Request
+	expandFields map[string]bool
+	selectFields []string
+}
+
+func newRequestDecorator(r *http.Request) *requestDecorator {
+	decoratedReq := &requestDecorator{
+		Request: r,
+	}
+
+	if expandFields, found := middleware.GetFieldsToExpand(r); found {
+		decoratedReq.expandFields = sliceToMap(expandFields)
+	}
+
+	if selectFields, found := middleware.GetFieldsToSelect(r); found {
+		decoratedReq.selectFields = selectFields
+	}
+
+	return decoratedReq
+}
+
+func sliceToMap(values []string) map[string]bool {
+	valueMap := make(map[string]bool, len(values))
+	for _, v := range values {
+		valueMap[v] = true
+	}
+	return valueMap
+}
+
+func (rd *requestDecorator) expands(field string) bool {
+	return rd.expandFields[field]
+}
+
+func (rd *requestDecorator) selects() []string {
+	return rd.selectFields
+}
+
+func (rd *requestDecorator) getVar(name string) string {
+	vars := mux.Vars(rd.Request)
+	return vars[name] // todo(sideninja) consider returning err if non-existing
+}
```

Comment:
instead of an error, you could just return the `found` parameter
```suggestion
func (rd *requestDecorator) getVar(name string) (string, bool) {
	vars := mux.Vars(rd.Request)
	value, found := vars[name]
	return value, found
}
```